package.json

```diff
@@ -91,7 +91,7 @@
     "typedoc": "^0.25.8",
     "typedoc-plugin-markdown": "^4.0.0-next.53",
     "typedoc-plugin-merge-modules": "^5.1.0",
-    "typescript": "5.1"
+    "typescript": "^5.4.5"
```

Comment:
upgraded TS because the latest version is much smarter with preserving narrowed types in callbacks, and that helped me avoid some awkward code structure in this PR